### PR TITLE
Update to 5.0 GA

### PR DESCRIPTION
**Pull request by dseefeld**

### CODE REVIEW

File: eng/Version.Details.xml
Comment (dagood) [inline eng/Version.Details.xml:78]: This is the wrong package, should be `Microsoft.AspNetCore.App.Runtime.win-x64` to avoid missing servicing updates.

File: patches/installer/0008-Update-blob-versions-to-use-ref-pkg-versions-to-trac.patch
Comment (dagood) [inline patches/installer/0008-Update-blob-versions-to-use-ref-pkg-versions-to-trac.patch:22]: This will break in servicing, the ref package doesn't update with every release but the runtime version will. I believe the infra we have putting them in a stable location is what's wrong here (or rather, not matching what the modern infra does anymore). The blob storage virtual folder should always be unstable and only the filename version gets stable.

File: repos/linker.proj
Comment (dagood) [inline repos/linker.proj:23]: Let both through with this bit I just noticed in other projects?

```xml
  <Target Name="SetOutputList" AfterTargets="Package" BeforeTargets="GatherBuiltPackages">
    <ItemGroup>
      <PackagesOutputList Include="$(ShippingPackagesOutput)" />
      <PackagesOutputList Include="$(NonShippingPackagesOutput)" />
    </ItemGroup>
  </Target>
```

File: repos/runtime.proj
Comment (dagood) [inline repos/runtime.proj:17]: What happens on a rebuild? What happens in bootstrapping when we build N using N? This seems pretty difficult to get right. Maybe it doesn't need to block merging this PR (so we can keep working on prebuilts in parallel ASAP) but I think we should find a less overwritey way immediately.

File: repos/linker.proj
Comment (dseefeld) [reply repos/linker.proj:23]: I'm not sure adding both is worthwhile, because later in the project, we're installing the version we built, which needs to use `PackagesOutput` to locate the package.

File: eng/Version.Details.xml
Comment (dseefeld) [reply eng/Version.Details.xml]: Updated.

File: patches/installer/0008-Update-blob-versions-to-use-ref-pkg-versions-to-trac.patch
Comment (dseefeld) [reply patches/installer/0008-Update-blob-versions-to-use-ref-pkg-versions-to-trac.patch:22]: Filed issue https://github.com/dotnet/source-build/issues/1853

**Comment by dseefeld** [comment]
@dagood @crummel  Any more comments on this?  I think it will be ready to merge as soon as it goes green.

File: repos/runtime.proj
Comment (dagood) [reply repos/runtime.proj]: Filed https://github.com/dotnet/source-build/issues/1854 for this.

**Comment by dagood** [comment]
Filed an issue for my remaining comment from earlier. Going through it again now.

File: Directory.Build.props
Comment (dagood) [inline Directory.Build.props:23]: Can we delete this line? #1627

File: tools-local/prebuilt-baseline-offline.xml
Comment (dagood) [inline tools-local/prebuilt-baseline-offline.xml:95]: Not sure if it'll cause a problem to remove this manually maintained section, but pointing it out.

**Review by dagood — APPROVED** [review]

File: patches/aspnetcore/0005-Fix-version-number.patch
Comment (crummel) [inline patches/aspnetcore/0005-Fix-version-number.patch:10]: Is this just not needed anymore or did the version number move somewhere else?

**Review by crummel — APPROVED** [review]
LGTM, just one question about the ASP.NET version.

File: patches/aspnetcore/0005-Fix-version-number.patch
Comment (dseefeld) [reply patches/aspnetcore/0005-Fix-version-number.patch:10]: The version number was 5.0.0.

File: Directory.Build.props
Comment (dseefeld) [reply Directory.Build.props:23]: No.  Without it, Roslyn wasn't building stable vesions.

File: tools-local/prebuilt-baseline-offline.xml
Comment (dseefeld) [reply tools-local/prebuilt-baseline-offline.xml:95]: Augh.  Missed that one.  Since the portable build is almost done, I'll wait and see if any of the portable prebuilts change and update it if so.

File: Directory.Build.props
Comment (dagood) [reply Directory.Build.props:23]: > Roslyn wasn't building stable versions.

Why is that a goal? The Microsoft Roslyn build didn't produce stable versions.
https://maestro-prod.westus2.cloudapp.azure.com/1297/https:%2F%2Fgithub.com%2Fdotnet%2Froslyn/68310/graph

> image

And in Version.Details.xml, `<Dependency Name="Microsoft.NETCore.Compilers" Version="3.8.0-5.20519.18" ...`

vs.
```
  [14:25:41.87] Building 'roslyn'...done
  New NuGet package(s) after building roslyn:
    -> Microsoft.CodeAnalysis.Build.Tasks 3.8.0
    -> Microsoft.CodeAnalysis.Common 3.8.0
```

Comment (dseefeld) [reply Directory.Build.props:23]: Hmm...  A later repo was looking for a stable version and wasn't finding it.  I was mistaken in my fix for that.

Comment (dagood) [reply Directory.Build.props:23]: I think it's fine to leave https://github.com/dotnet/source-build/issues/1627 tracking it for followup but wanted to dig into any issues to make it easier to tackle later.

File: patches/aspnetcore/0005-Fix-version-number.patch
Comment (dagood) [reply patches/aspnetcore/0005-Fix-version-number.patch:10]: I take it you mean that the version number already present in the file is 5.0.0, so we don't need to patch it to 5.0.0? Maybe we should open an issue to track adding this patch back in 5.0.1 when the version gets desynchronized again, so we're less likely to forget/rediscover it in the future?

Comment (dseefeld) [reply patches/aspnetcore/0005-Fix-version-number.patch:10]: Yes.  It didn't need to be updated.  I added https://github.com/dotnet/source-build/issues/1855

File: Directory.Build.props
Comment (dseefeld) [reply Directory.Build.props:23]: Without the patch, the production build produces `Microsoft.NETCore.Compilers.3.8.0-dev.20519.18.nupkg`.  The "dev" version doesn't seem right.

Comment (dagood) [reply Directory.Build.props:23]: That sounds like a missing `-ci` or `OfficialBuildId` or something along those lines.

Comment (dseefeld) [reply Directory.Build.props:23]: I didn't change roslyn.proj.  It has `OfficialBuild=true` here: https://github.com/dotnet/source-build/blob/92d2aefc755e365d09f661fbaf9e0c3755f2179d/repos/roslyn.proj#L12

Comment (dseefeld) [reply Directory.Build.props:23]: Adding `--ci` to roslyn.proj caused it to generate the desired version.  `Microsoft.Net.Compilers.Toolset.3.8.0-5.20519.18.nupkg`  I'll fix this once this PR is merged under #1627

File: patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch
Comment (tmds) [inline patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch:11]: @dseefeld @dagood does this mean something doesn't allow us to consume `System.Security.Cryptography.Cng` `5.0.0` to produce `System.Security.Cryptography.Cng` `5.0.0`?

Comment (dseefeld) [reply patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch:11]: Yes, that's what I was seeing.  Here's the error I was seeing without this patch:
```
3000 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error : An item with the same key has already been added. Key: (System.Security.Cryptography.Cng, 5.0.0) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3001 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3002 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at System.Linq.Enumerable.ToDictionary[TSource,TKey](List`1 source, Func`2 keySelector, IEqualityComparer`1 comparer) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3003 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Commands.LockFileBuilder.CreateLockFile(LockFile previousLockFile, PackageSpec project, IEnumerable`1 targetGraphs, IReadOnlyList`1 localRepositories, RemoteWalkContext context) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3004 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Commands.RestoreCommand.ExecuteAsync(CancellationToken token) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3005 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Commands.RestoreRunner.ExecuteAsync(RestoreSummaryRequest summaryRequest, CancellationToken token) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3006 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Commands.RestoreRunner.ExecuteAndCommitAsync(RestoreSummaryRequest summaryRequest, CancellationToken token) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3007 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Commands.RestoreRunner.CompleteTaskAsync(List`1 restoreTasks) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3008 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Commands.RestoreRunner.RunAsync(IEnumerable`1 restoreRequests, RestoreArgs restoreContext, CancellationToken token) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3009 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Commands.RestoreRunner.RunAsync(RestoreArgs restoreContext, CancellationToken token) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3010 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Build.Tasks.BuildTasksUtility.RestoreAsync(DependencyGraphSpec dependencyGraphSpec, Boolean interactive, Boolean recursive, Boolean noCache, Boolean ignoreFailedSources, Boolean disableParallel, Boolean force, Boolean forceEvaluate, Boolean hideWarningsAndErrors, Boolean restorePC, Boolean cleanupAssetsForUnsupportedProjects, ILogger log, CancellationToken canc     ellationToken) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]
3011 /src/tb-5.0GA/.dotnet/sdk/5.0.100-rc.1.20452.10/NuGet.RestoreEx.targets(10,5): error :    at NuGet.Build.Tasks.Console.MSBuildStaticGraphRestore.RestoreAsync(String entryProjectFilePath, IDictionary`2 globalProperties, IReadOnlyDictionary`2 options) [/src/tb-5.0GA/src/runtime.cf258a14b70ad9069470a108f13765e0e5988f51/Build.proj]

```

Comment (tmds) [reply patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch:11]: Doesn't this happen for a lot of packages since we're consuming 5.0.0 prebuilt packages to build 5.0.0 source-built packages? Or is issue somehow specific to the Cng package?

Comment (dseefeld) [reply patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch:11]: I don't know why this issue popped up for this specific package.  We could certainly try to remove the patch and see if the issue repros.  It may have been a "point in time" issue.

Comment (dagood) [reply patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch:11]: Here's some more info from our old chat about it. dotnet/runtime seems to be doing something specific that aggravates some NuGet bug. It won't be a point in time issue, I doubt the NuGet issue will be fixed in our timeline. I was able to put together a simplified repro, but we dropped it because we probably don't actually need a fix here. (And we always want to cut things/work because time is not something we have much of.)

> dagood: [in response to a binlog] Yeah, good old RestoreTaskEx... another issue we have with static restore is https://github.com/dotnet/source-build/issues/1825 FWIW, but unrelated.
> 
> ...
>
> Seems like we might be running into https://github.com/NuGet/Home/issues/6795 ? But not sure why we would but the Microsoft build wouldn't...

> [10/30 9:51 AM] Davis Goodin
>     Hmm, noticed a few things earlier in those RestoreTaskEx logs:
> 
> ```
> Could not parse version  for LatestPackage System.Security.Cryptography.Cng, will use latest stable.
> ...
> Installing System.Security.Cryptography.Cng 5.0.0.
> ```
> 
> (And actually the lines do seem scrambled around that installing line, so not sure which project it's coming from.)

> ​[10/30 9:52 AM] Davis Goodin
>     I guess some other project is referring to the Cng package. Not totally sure that's important though.

> ​[10/30 9:55 AM] Davis Goodin
>     Hmm. Wonder if it's choking on this, where in the prod build we don't have a PVP value but in the tarball build we're overwriting it?
> https://github.com/dotnet/runtime/blob/7ef6d50b312217d2f7c17b9697891fa8ab98a19d/src/libraries/System.Security.Cryptography.Pkcs/src/System.Security.Cryptography.Pkcs.csproj#L639-L642
> 
> ```
>  <ItemGroup Condition="$(TargetFramework.StartsWith('netstandard2.0'))">
>  <PackageReference Include="System.Buffers" Version="$(SystemBuffersVersion)" />
>  <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
>  <PackageReference Include="System.Security.Cryptography.Cng" Version="$(SystemSecurityCryptographyCngVersion)" />
> ```
> 

> ​[10/30 9:57 AM] Davis Goodin
>     For a sanity check... does the PVP being passed in in your local build have SystemSecurityCryptographyCngVersion defined?

> [10/30 9:58 AM] Dan Seefeldt
>     Yes.
> ​[10/30 9:59 AM] Dan Seefeldt
>     The difference between production and tarball is that production build has it defined as 4.7.0 while the tarball build is 5.0.0.

> [10/30 10:40 AM] Davis Goodin
>   Got it, looks like TargetFrameworks is part of the key for NuGet to mash them together and fail: https://github.com/dagood/repro/tree/repro-static-restore-ref-mashup
> 
> Oh huh. Static restore is not necessary to repro.
>
> (I was wondering if we could also fix it by disabling static restore.)
>
> After mulling over what Tom's suggesting (and Wes suggested in the past) I think the real long term fix is to stop building all TFMs except net5.0... (so that we don't have this PackageReference to begin with) [referring to #1763]

> dseefeld: BTW, forcing System.Security.Cryptography.Cng to 4.7.0 fixed the issue.

4.7.0 isn't a problem because we should be able to build it in SBRP, so we should be able to leave it at that.

This info probably should have made it to the patch, or an issue linked to from the patch. (But we're rushing.)

Comment (tmds) [reply patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch:11]: @dagood since you have a small reproducer, it seems worth filing a bug in https://github.com/NuGet/Home (even if it won't be fixed for our 5.0).

Comment (dagood) [reply patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch:11]: Agreed, but I don't feel I have time to make it make enough sense for discussion. Filed https://github.com/dotnet/source-build/issues/1888 to track it though.

Comment (dagood) [reply patches/runtime/0026-Pin-System.Security.Cryptography.Cng-to-single-versi.patch:11]: FYI, I've filed the issue for this: https://github.com/NuGet/Home/issues/10368.

**Pull request merged**
